File: C_Examples/spawn_2.c
/**
 * Spawn_2.c : A program to spawn two cpu bound threads 
 * Daniel Zahka ~ daniel.zahka@wustl.edu
 * Anish Naik ~ anish,r.naik@wustl.edu
 *
 */

#include <stdlib.h>
#include <pthread.h>
#include <stdio.h>

#define ITERATIONS (10000000000)

void * thread_func(void * args){
    long long i, k;
    for (i = 0; i < ITERATIONS; ++i){
        k = i * 3;
    }
    return 0;
}


int main(){
    /*Check return values maybe...*/
    pthread_t tid1, tid2;
    printf("Spawning Threads\n");
    pthread_create(&tid1, 0, thread_func, 0); 
    pthread_create(&tid2, 0, thread_func, 0);
    pthread_join(tid1, 0);
    pthread_join(tid2, 0); 
    printf("Joined Threads\n");
 
    return 0;
}

File: py_examples/spawn_2.py
"""
spawn_2.py : Spawns two native cpu bound threads

Daniel Zahka ~ daniel.zahka@wustl.edu
Anish Naik ~ anish.r.naik@wustl.edu
"""

import threading
import time

def thread_func():
        for i in xrange(1, 50000000):
                k = i * 3


if __name__ == "__main__":
    t1 = threading.Thread(target=thread_func, name="spawn_2_t1")
    t2 = threading.Thread(target=thread_func, name="spawn_2_t1")
    print("Spawning Threads...")
    start = time.time()
    t1.start()
    t2.start()
    t1.join()
    t2.join()
    end = time.time()
    print("Joined Threads...")
    print(end - start)

